### PR TITLE
TINY-11332: Update dependencies

### PR DESCRIPTION
**Pull request by jscasca**
Related Ticket: TINY-11332

Description of Changes:
* Update dompurify and relevant tests for security patch

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] Docs ticket created (if applicable)

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/src/core/main/ts/html/Sanitization.ts [hunk 1]
```diff
@@ -178,14 +178,16 @@ const setupPurify = (settings: DomParserSettings, schema: Schema, namespaceTrack
...
+  const basePurifyConfig: Config & { SAFE_FOR_XML: boolean } = {
     IN_PLACE: true,
     ALLOW_UNKNOWN_PROTOCOLS: true,
     // Deliberately ban all tags and attributes by default, and then un-ban them on demand in hooks
     // #comment and #cdata-section are always allowed as they aren't controlled via the schema
     // body is also allowed due to the DOMPurify checking the root node before sanitizing
     ALLOWED_TAGS: [ '#comment', '#cdata-section', 'body' ],
-    ALLOWED_ATTR: []
+    ALLOWED_ATTR: [],
+    // TINY-11332: New settings for dompurify 3.1.7
```
Comment (spocke) [inline modules/tinymce/src/core/main/ts/html/Sanitization.ts:189]: Could you add some details that this is not yet available in the types so maybe someone spots that and fixes in once they do update the types.

File: modules/tinymce/src/core/test/ts/browser/content/insert/NamespaceElementsTest.ts [hunk 2]
```diff
@@ -11,12 +11,13 @@ describe('browser.tinymce.core.content.insert.NamespaceElementsTest', () => {
...
 
-  it('TINY-10237: Inserting SVG elements but filter out things like scripts', () => {
+  // Updated for TINY-11332: Remove html inside SVG
+  it('TINY-10237: Inserting SVG elements but filter out things like scripts and invalid children', () => {
     const editor = hook.editor();
     editor.setContent('<p>ab</p>');
     TinySelections.setCursor(editor, [ 0, 0 ], 1);
     editor.insertContent('<svg><circle cx="50" cy="50" r="40" stroke="black" stroke-width="3" fill="red"><desc><script>alert(1)</script><p>hello</p></circle></a></svg>');
-    TinyAssertions.assertContent(editor, '<p>a<svg><circle cx="50" cy="50" r="40" stroke="black" stroke-width="3" fill="red"><desc><p>hello</p></desc></circle></svg>b</p>');
+    TinyAssertions.assertContent(editor, '<p>a<svg><circle cx="50" cy="50" r="40" stroke="black" stroke-width="3" fill="red"><desc></desc></circle></svg>b</p>');
```
Comment (spocke) [inline modules/tinymce/src/core/test/ts/browser/content/insert/NamespaceElementsTest.ts:20]: We could still check if text is properly retained in `desc`.

File: modules/tinymce/src/core/test/ts/browser/content/insert/NamespaceElementsTest.ts [hunk 3]
```diff
@@ -11,12 +11,13 @@ describe('browser.tinymce.core.content.insert.NamespaceElementsTest', () => {
     valid_children: '+p[math]'
   }, [], true);
 
-  it('TINY-10237: Inserting SVG elements but filter out things like scripts', () => {
+  // Updated for TINY-11332: Remove html inside SVG
```
Comment (spocke) [inline modules/tinymce/src/core/test/ts/browser/content/insert/NamespaceElementsTest.ts:14]: I wonder if we need these comments and have a separate test that just validates this specific new behaviour and update the old tests to not have anything but scripts and text in the input data?

File: .changes/unreleased/tinymce-TINY-11332-2024-10-08.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+project: tinymce
+kind: Fixed
+body: 'Invalid HTML elements within an `svg` element are removed. #TINY-1132'
```
Comment (spocke) [inline .changes/unreleased/tinymce-TINY-11332-2024-10-08.yaml:3]: Maybe this:

```suggestion
body: 'Invalid HTML elements within SVG elements was not removed.
```

Or:
```suggestion
body: 'Invalid HTML elements within `svg` elements was not removed.
```

Or:
```suggestion
body: 'Invalid HTML elements within `svg` elements is now removed.
```

**Review by spocke — APPROVED** [review]

File: modules/tinymce/src/core/test/ts/browser/content/insert/NamespaceElementsTest.ts (hunk 2 above)
Comment (jscasca) [reply modules/tinymce/src/core/test/ts/browser/content/insert/NamespaceElementsTest.ts:19]: Added new test for keeping content

File: modules/tinymce/src/core/main/ts/html/Sanitization.ts (hunk 1 above)
Comment (jscasca) [reply modules/tinymce/src/core/main/ts/html/Sanitization.ts:190]: Added comments regarding types and versions available

File: modules/tinymce/src/core/test/ts/browser/content/insert/NamespaceElementsTest.ts (hunk 3 above)
Comment (jscasca) [reply modules/tinymce/src/core/test/ts/browser/content/insert/NamespaceElementsTest.ts]: Removed comments and added new test

File: .changes/unreleased/tinymce-TINY-11332-2024-10-08.yaml (hunk 4 above)
Comment (jscasca) [reply .changes/unreleased/tinymce-TINY-11332-2024-10-08.yaml]: Updated changelog entry

**Review by shanmen-tiny — APPROVED** [review]

**Review by ArvinJ-H — APPROVED** [review]

**Pull request merged**